src/python/test/test_dxclient.py

```diff
@@ -2781,36 +2782,87 @@ def _get_analysis_id(dx_run_output):
-        watched_job_desc = dxpy.describe(watched_job_id)
-        self.assertEqual(watched_job_desc['applet'], applet_id)
-        self.assertEqual(watched_job_desc['priority'], 'high')
+        ssh_job_id = dx_run_output.split('\n')[0]
+        ssh_job_desc = dxpy.describe(ssh_job_id)
+        self.assertEqual(ssh_job_desc['applet'], applet_id)
+        self.assertEqual(ssh_job_desc['priority'], 'high')
 
         # don't actually need it to run
-        run("dx terminate " + watched_job_id)
+        run("dx terminate " + ssh_job_id)
 
-        # --allow-ssh implies --priority high
+        # --allow-ssh implies high priority when --priority is not specified
         try:
             dx_run_output = run("dx run myapplet -y --allow-ssh --brief")
         except subprocess.CalledProcessError:
             # ignore any ssh errors; just want to test requested
             # priority
             pass
-        watched_job_id = dx_run_output.split('\n')[0]
+        allow_ssh_job_id = dx_run_output.split('\n')[0]
+        allow_ssh_job_desc = dxpy.describe(allow_ssh_job_id)
+        self.assertEqual(allow_ssh_job_desc['applet'], applet_id)
+        self.assertEqual(allow_ssh_job_desc['priority'], 'high')
+
+        # don't actually need it to run
+        run("dx terminate " + allow_ssh_job_id)
+
+        # warning when --priority is normal/low with --watch
+        try:
+            watched_run_output = run("dx run myapplet -y --watch --priority normal")
+        except subprocess.CalledProcessError:
+            # ignore any watch errors; just want to test requested
+            # priority
+            pass
+        watched_job_id = re.search('job-[A-Za-z0-9]{24}', watched_run_output).group(0)
         watched_job_desc = dxpy.describe(watched_job_id)
         self.assertEqual(watched_job_desc['applet'], applet_id)
-        self.assertEqual(watched_job_desc['priority'], 'high')
+        self.assertEqual(watched_job_desc['priority'], 'normal')
+        for string in ["WARNING", "normal", "interrupting interactive work"]:
+            self.assertIn(string, watched_run_output)
 
         # don't actually need it to run
         run("dx terminate " + watched_job_id)
 
+        # no warning when --brief and --priority is normal/low with --allow-ssh
+        try:
+            allow_ssh_run_output = run("dx run myapplet -y --allow-ssh --priority normal --brief")
+        except subprocess.CalledProcessError:
+            # ignore any ssh errors; just want to test requested
+            # priority
+            pass
+        allow_ssh_job_id = re.search('job-[A-Za-z0-9]{24}', allow_ssh_run_output).group(0)
+        allow_ssh_job_desc = dxpy.describe(allow_ssh_job_id)
+        self.assertEqual(allow_ssh_job_desc['applet'], applet_id)
+        self.assertEqual(allow_ssh_job_desc['priority'], 'normal')
+        for string in ["WARNING", "normal", "interrupting interactive work"]:
+            self.assertNotIn(string, allow_ssh_run_output)
```

Comment:
Can this be moved under try? This should fix the failing test. Then run it in a branch on jenkins if possible.

Reply:
Sure that is a nice solution. Will do it and let you check :)